### PR TITLE
feat(catalog): add `spec.target(s)` for Location entities at the `CatalogTable`

### PR DESCRIPTION
**Pull request by pjungermann**
Signed-off-by: Patrick Jungermann <Patrick.Jungermann@gmail.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by pjungermann** [comment]
Screen Shot 1
Screen Shot 2

**Comment by JordiPolo** [comment]
Not sure what's `target` is. doesn't the action to view the source already link to it? According to the screenshot, it does take a lot of real estate in the table as if it were something super important.

**Comment by pjungermann** [comment]
@JordiPolo The `Location` entity has `spec.target` or `spec.targets` which point to catalog files which get read by the processing loop.

> doesn't the action to view the source already link to it?

The "source view" link points to the source dir which contains those files. But this is for the entities which resulted from reading those targets (the catalog files they reference). At `Location` entities - if available - it would point to the dir which contains the catalog file itself originated from, not the target(s).

At the moment, there is no easy way to find which targets (catalog files) a particular `Location` is about.

Let's take the demo as example:
https://demo.backstage.io/catalog?filters%5Bkind%5D=location&filters%5Buser%5D=all

https://demo.backstage.io/catalog/default/location/example-apis
=>
You need to open the entity page and then go to the "..." menu, open "Inspect entity" and switch to "Raw YAML" or "Raw JSON" when you want find them.

To address this I've created PR #12671 to add this straight to the About Card of the `Location` entity page.

If you have thousands of `Location` entities, going to all of them is not great though. This PR will add this info to the table incl. allowing to filter by target (e.g., part of the URL/path/reference).

> it does take a lot of real estate in the table as if it were something super important

I would argue that this info the most important of a `Location` entity.

`name` is usually generated, type is mostly `url` or sometimes `file` (depending on the setup), and `description` and `tags` are usually empty.

In the screenshot, you can see an example with `description` being present. This is just some example data and is rare. All `Location` entities created/generated by processors or providers will not contain this.

**Comment by kissmikijr** [comment]
Love this! 
Actually, I think the target is almost the most important property of a Location, since for the generated ones their names are hashed you cannot really find the one you are looking for unless you open the raw yaml and check there...

File: plugins/catalog/src/components/CatalogTable/columns.tsx [hunk 1]
```diff
@@ -82,11 +83,31 @@ export const columnFactories = Object.freeze({
...
+            <OverflowTooltip
+              text={(
+                (entity!.spec!.targets as JsonArray) || [entity.spec.target]
+              ).join(', ')}
+              placement="bottom-start"
+            />
+          )}
+        </>
+      ),
+      // width: 'calc((100% - 0px) / 3)',
```
Comment (benjdlambert) [inline plugins/catalog/src/components/CatalogTable/columns.tsx:102]: What should we do with this here?

**Review by benjdlambert — COMMENTED** [review]
Overall think this is a nice change, just a question about the widths :pray:

File: plugins/catalog/src/components/CatalogTable/columns.tsx (hunk 1 above)
Comment (pjungermann) [reply plugins/catalog/src/components/CatalogTable/columns.tsx:102]: my bad. I will remove it

**Pull request merged**